rust/src/examples.rs

```diff
@@ -0,0 +1,214 @@
+/*!
```

Comment:
there are some broken links in this file:
```
warning: `[ApiSettings]` cannot be resolved, ignoring it.
   --> src/examples.rs:112:26
    |
112 | Going from the top, the [`ApiSettings`] include [`bind_address`], which is the
    |                          ^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = note: `#[warn(intra_doc_link_resolution_failure)]` on by default
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[bind_address]` cannot be resolved, ignoring it.
   --> src/examples.rs:112:50
    |
112 | Going from the top, the [`ApiSettings`] include [`bind_address`], which is the
    |                                                  ^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[PetSettings]` cannot be resolved, ignoring it.
   --> src/examples.rs:116:6
    |
116 | The [`PetSettings`] specify various parameters of the PET protocol.
    |      ^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[sum]` cannot be resolved, ignoring it.
   --> src/examples.rs:119:16
    |
119 | important are [`sum`] and [`update`], which are the probabilities assigned to
    |                ^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[update]` cannot be resolved, ignoring it.
   --> src/examples.rs:119:28
    |
119 | important are [`sum`] and [`update`], which are the probabilities assigned to
    |                            ^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[min_sum_count]` cannot be resolved, ignoring it.
   --> src/examples.rs:123:17
    |
123 | - The settings [`min_sum_count`] and [`min_update_count`] specify, respectively,
    |                 ^^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[min_update_count]` cannot be resolved, ignoring it.
   --> src/examples.rs:123:39
    |
123 | - The settings [`min_sum_count`] and [`min_update_count`] specify, respectively,
    |                                       ^^^^^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[min_sum_time]` cannot be resolved, ignoring it.
   --> src/examples.rs:128:32
    |
128 | - To complement, the settings [`min_sum_time`] and
    |                                ^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[min_update_time]` cannot be resolved, ignoring it.
   --> src/examples.rs:129:2
    |
129 | [`min_update_time`] specify, respectively, the minimum amount of time (in
    |  ^^^^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[expected_participants]` cannot be resolved, ignoring it.
   --> src/examples.rs:133:4
    |
133 | - [`expected_participants`] should be set to the expected number of participants
    |    ^^^^^^^^^^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[MaskSettings]` cannot be resolved, ignoring it.
   --> src/examples.rs:136:6
    |
136 | The [`MaskSettings`] determines the masking configuration, consisting of the
    |      ^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[ModelSettings]` cannot be resolved, ignoring it.
   --> src/examples.rs:137:56
    |
137 | group type, data type, bound type and model type. The [`ModelSettings`] specify
    |                                                        ^^^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

warning: `[Client]` cannot be resolved, ignoring it.
   --> src/examples.rs:201:66
    |
201 | The example creates the expected number of participants (called [`Client`]s
    |                                                                  ^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`
```

Reply:
added the links